### ISSUE TITLE
2.1.1 On shoot event

### ISSUE THREAD

**Issue by G1R-Productions** [Njol/Skript#312]
The version skript 2.1.1 shoot even doesn't seem to work. While in th previous version 2.0.2 it works perfectly.

I tested this by doing a simple:

on shoot:
     send "Test" to shooter

This skript works only on 2.0.2 and not 2.1.1 though no errors pop up just the plugin not registering the shoot even.

**Comment by G1R-Productions** [comment]
This happens when I shoot a pig:
http://pastebin.com/BFFedZ6s

**Comment by G1R-Productions** [comment]
Also tested with
on shoot:
           execute command "give ghaz_ranka 1 1"

Along with:

on damage:
            send "test" to attacker

When I swing a sword at the target I get sent "test" but if I shoot an arrow I get nothing.

as well as:
on shoot:
            cancel event

**Comment by G1R-Productions** [comment]
EDIT:

on shoot:
cancel event
send "test" to shooter

the cancel event works but it took like an hour after creating and loading the skript for it to work but the send to shooter doesn't work -- so the issue may be that the "shooter" is not registering correctly.

**Comment by XHawk87** [comment]
This is a serious issue with Skript 2.1.2. It is caused by using ProjectileSource which only exists in the current dev builds of CraftBukkit. It is not backwards compatible to earlier versions of CraftBukkit which many servers need to remain on. It has broken everything to do with projectiles in Skript.

**Comment by ismooch** [comment]
I understand that it is an issue @XHawk87 that should have been looked at because the latest stable build does not support it. Though, the last few Dev Builds of CraftBukkit support this just fine. As does the latest spigot build. There are many ways to fix this, I would hardly call it a 'serious' issue, as I do not know any plugins that have lost support with the latest dev build.

**Comment by XHawk87** [comment]
There are many plugins available that use reflection or build against CraftBukkit so cannot be immediately updated. There are also servers running mods through MCPC+ or BukkitForge which cannot be updated to later versions. What is more, the CB 1.7.5 snapshots are provided for testing purposes only, so it is unreasonable to expect every server to update to using them.

**Comment by ismooch** [comment]
the 1.7.2 R3 and R4 work just fine with this. I do not use 1.7.5 and everything works just fine with on shoot events for me, in fact I ran into this EXACT problem with R2 Bukkit builds.. and just pushed to R4, everything was gravy after that.

**Comment by XHawk87** [comment]
It could be a long time before a stable or even beta release is made that contains the ProjectileSource update. The dev builds are just that, for development, you cannot expect people to use them unless they are assisting in development. Betas are for beta-testing, you still cannot expect people to use them if they do not want to participate in beta-testing. People may choose to use them at their own risk, but it is their risk to take, not something they should be pressured into. 

Using features that exist only in a dev build of Bukkit should be marked as a dev update of Skript and should include a warning to users of this requirement, as well as appropriate game versions being listed. Skript 2.1.2 claims to work from versions 1.7.2-R0.3 way back to 1.2.5. You cannot make that claim without backing it up in code.

**Comment by ismooch** [comment]
@xhawk87 You're right. I often wrongly assume that everyone running a server would take the same course as I would, and thats silly of me. Skript does openly announce that it is backward compatible. If it is not, it should definitely be corrected, or at least reflect the fact that it is not. I agree with you.
